packages/language-server/src/server.ts

```diff
@@ -174,3 +225,60 @@ documents.onDidClose((e) => {
 documents.listen(connection);
 
 connection.listen();
+
+function isManifestDoc(uri: string): boolean {
+  return uri.endsWith("manifest.json");
+}
+
+function getFlagForXMLFile(uri: string): boolean {
+  const manifestFilesForCurrentFolder = Object.keys(
+    manifestData
+  ).filter((manifestUri) => uri.includes(dirname(manifestUri)));
+  const requiredManifestPath = maxBy(
+    manifestFilesForCurrentFolder,
+    (manifestFile) => manifestFile.length
+  );
+
+  if (requiredManifestPath === undefined) {
+    return false;
+  }
+
+  return manifestData[requiredManifestPath].isFlexEnabled;
+}
+
+async function getManifestDocumentsPromise(): Promise<string[]> {
```

Comment:
The name `getManifestDocumentsPromise` is too generic.
the `get` / `set` prefixes should be used mainly for simple getter/setters.
But here there is more advanced logic. in our case we are **scanning/reading/finding** all the manifest.json files. 

So the name of the function should reflect that.

also the suffix `promise` is redundant, its async and returns Promise of string[].